src/skai/model/data.py

```diff
@@ -177,6 +177,215 @@ def apply_batch(dataloader, batch_size):
   return dataloader
 
 
+class DataEncoder:
+  def __init__(self):
+    string_label_categories: dict[str, int] = {b'bad_example' :0,
+                                    b'destroyed'   :1,
+                                    b'major_damage':2,
+                                    b'minor_damage':3,
+                                    b'no_damage'   :4}
+    self.label_to_int_table = tf.lookup.StaticHashTable(
+      tf.lookup.KeyValueTensorInitializer(
+        list(string_label_categories.keys()),
+        list(string_label_categories.values())),
+        default_value=-1
+      )
+    self.int_to_label_table = tf.lookup.StaticHashTable(
+      tf.lookup.KeyValueTensorInitializer(
+        list(string_label_categories.values()),
+        list(string_label_categories.keys())),
+        default_value='unknown'
+      )
+  def encode_example_ids(self, dataloader: Dataloader)-> Dataloader:
```

Comment:
My understanding is that the only place we use example_ids during training is for the bias table. If so, I don't see a reason why we can't just use the int64_id feature that is already included in each example (see https://github.com/google-research/skai/blob/13b6bc58e088180c1b89af1bfa0103494270e5a2/src/skai/generate_examples.py#L452) instead of example_id. That would save a lot of the complexity of converting the example_ids into ints and back on the fly here. WDYT @panford @jlee24 ?